### PR TITLE
Feature/dope 190 any every follow up

### PR DESCRIPTION
**Pull request by jansigi**

### CODE REVIEW

**Review by martinagallati-ergon — COMMENTED** [review]
Seems good! I saw you implemented both `ANY` and `EVERY` separately, what about `ANY AND EVERY ... IN ... SATISFIES`?

**Comment by jansigi** [comment]
@martinagallati-ergon I created a ticket: dope-253

**Review by martinagallati-ergon — APPROVED** [review]

File: core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/collection/SatisfiesExpression.kt [hunk 1]
```diff
@@ -27,13 +29,13 @@ enum class SatisfiesType {
...
-    private val variable: String,
+    private val arrayExpression: TypeExpression<ArrayType<T>>,
+    private val iteratorName: String,
     private val predicate: (Iterator<T>) -> TypeExpression<BooleanType>,
 ) : TypeExpression<BooleanType> {
     override fun toDopeQuery(): DopeQuery {
-        val listDopeQuery = list.toDopeQuery()
-        val iteratorVariable = if (variable == DEFAULT_ITERATOR_VARIABLE) variable + IteratorManager.count else variable
+        val listDopeQuery = arrayExpression.toDopeQuery()
+        val iteratorVariable = if (iteratorName == DEFAULT_ITERATOR_VARIABLE) IteratorManager.getIteratorName() else iteratorName
```
Comment (pgruntz) [inline core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/collection/SatisfiesExpression.kt:38]: This is a but stupid, you misuse the constant DEFAULT_ITERATOR_VARIABLE here just to check if the user entered an iteratorName. As dicussed offline, just make it nullable

**Review by pgruntz — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/collection/SatisfiesExpression.kt (hunk 1 above)
Comment (jansigi) [reply core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/collection/SatisfiesExpression.kt]: done

**Review by pgruntz — APPROVED** [review]

**Pull request merged**